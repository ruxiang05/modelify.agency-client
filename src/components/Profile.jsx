import React from 'react';
import PropTypes from 'prop-types';
import { UserContext } from '../contexts/userContext';

class Profile extends React.Component {
  constructor(props) {
    super(props);

    this.logout = this.logout.bind(this);
  }

  logout() {
    const { history } = this.props;
    const { handleLogout } = this.context;
    handleLogout();
    history.push('/login');
  }

  render() {
    return (
      <button type="button" onClick={this.logout}>
                Logout
      </button>
    );
  }
}
Profile.propTypes = {
  history: PropTypes.shape({}).isRequired,
};
Profile.contextType = UserContext;
export default Profile;
