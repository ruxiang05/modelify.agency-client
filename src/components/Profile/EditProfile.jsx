/* Uses methods and/or components from react and prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import FormInput from '../FormInput';
import api from '../../api';
import { getToken, setToken } from '../../auth';
import PageHeader from '../PageHeader';

class EditProfile extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: props.email,
      name: props.name,
      role: props.role,
      phoneNumber: props.phoneNumber,
      dateOfBirth: props.modelInfo.dateOfBirth,
      address: props.modelInfo.address,
      eyes: props.modelInfo.characteristics.eyes,
      hair: props.modelInfo.characteristics.hair,
      skin: props.modelInfo.characteristics.skin,
      height: props.modelInfo.measurements.height,
      weight: props.modelInfo.measurements.weight,
      chest: props.modelInfo.measurements.chest,
      waist: props.modelInfo.measurements.waist,
      hips: props.modelInfo.measurements.hips,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.formatModel = this.formatModel.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();
    const {
      email, name, role, phoneNumber, ...modelInfo
    } = this.state;
    let user = {
      email, name, role, phoneNumber,
    };
    if (role === 'model') {
      user = { ...user, modelInfo: this.formatModel(modelInfo) };
    }
    const { toggleEdit, updateUser } = this.props;
    api.users.editProfile(getToken(), user).then((data) => {
      setToken(data.token);
      updateUser();
      toggleEdit();
    });
  }

  formatModel(model) {
    const {
      dateOfBirth, address, eyes, hair, skin, height, weight, chest, waist, hips,
    } = model;
    return {
      dateOfBirth,
      address,
      characteristics: {
        eyes,
        hair,
        skin,
      },
      measurements: {
        height,
        weight,
        chest,
        waist,
        hips,
      },
    };
  }

  handleChange(event) {
    const { name, value } = event.target;
    this.setState(prevState => ({ ...prevState, [name]: value }));
  }

  render() {
    const {
      email, name, phoneNumber, role, dateOfBirth, address, eyes, hair, skin, height, weight, chest, waist, hips,
    } = this.state;
    const { toggleEdit } = this.props;
    return (
      <React.Fragment>
        <PageHeader title="Edit" backButton goBackAction={toggleEdit} />

        <div className="page">
          <form onSubmit={this.handleSubmit}>
            <h2>Account Details</h2>
            <FormInput
              type="text"
              name="email"
              title="Email"
              value={email}
              placeholder="Enter your email"
              handleChange={this.handleChange}
            />
            <FormInput
              type="text"
              name="name"
              title="Name"
              value={name}
              placeholder="Enter your name"
              handleChange={this.handleChange}
            />
            <FormInput
              type="text"
              name="phoneNumber"
              title="Phone"
              value={phoneNumber}
              placeholder="Enter your phone number"
              handleChange={this.handleChange}
            />
            {role === 'model' && (
            <React.Fragment>
              <p>Model Details</p>
              <FormInput
                type="date"
                name="dateOfBirth"
                title="Date Of Birth"
                value={dateOfBirth}
                handleChange={this.handleChange}
              />
              <FormInput
                type="text"
                name="address"
                title="Enter your address"
                value={address}
                placeholder="Address"
                handleChange={this.handleChange}
              />
              <h2>Characteristics</h2>
              <FormInput
                type="text"
                name="eyes"
                title="Eye Colour"
                value={eyes}
                placeholder="Enter your eye colour"
                handleChange={this.handleChange}
              />
              <FormInput
                type="text"
                name="hair"
                title="Hair colour"
                value={hair}
                placeholder="Enter your hair Colour"
                handleChange={this.handleChange}
              />
              <FormInput
                type="text"
                name="skin"
                title="Skin Colour"
                value={skin}
                placeholder="Enter your skin"
                handleChange={this.handleChange}
              />
              <h2>Measurements</h2>
              <FormInput
                type="text"
                name="height"
                title="Height"
                value={height}
                placeholder="Enter our height"
                handleChange={this.handleChange}
              />
              <FormInput
                type="text"
                name="weight"
                title="Weight"
                value={weight}
                placeholder="Enter our weight"
                handleChange={this.handleChange}
              />
              <FormInput
                type="text"
                name="chest"
                title="Chest"
                value={chest}
                placeholder="Enter our chest size"
                handleChange={this.handleChange}
              />
              <FormInput
                type="text"
                name="waist"
                title="Waist"
                value={waist}
                placeholder="Enter our waist size"
                handleChange={this.handleChange}
              />
              <FormInput
                type="text"
                name="hips"
                title="Hips"
                value={hips}
                placeholder="Enter your hip size"
                handleChange={this.handleChange}
              />
            </React.Fragment>
            )}
            <div className="form-submit">
              <input type="submit" value="Done" />
            </div>
          </form>
        </div>
      </React.Fragment>
    );
  }
}

EditProfile.propTypes = {
  email: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  phoneNumber: PropTypes.string.isRequired,
  modelInfo: PropTypes.shape({
    dateOfBirth: PropTypes.string.isRequired,
    address: PropTypes.string.isRequired,
    characteristics: PropTypes.shape({
      eyes: PropTypes.string.isRequired,
      hair: PropTypes.string.isRequired,
      skin: PropTypes.string.isRequired,
    }),
    measurements: PropTypes.shape({
      height: PropTypes.string.isRequired,
      weight: PropTypes.string.isRequired,
      chest: PropTypes.string.isRequired,
      waist: PropTypes.string.isRequired,
      hips: PropTypes.string.isRequired,
    }),
  }).isRequired,
  role: PropTypes.string.isRequired,
  toggleEdit: PropTypes.func.isRequired,
  updateUser: PropTypes.func.isRequired,
};

export default EditProfile;
