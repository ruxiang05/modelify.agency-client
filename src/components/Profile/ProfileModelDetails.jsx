import React from 'react';
import PropTypes from 'prop-types';
import '../../styles/profile-model-details.scss';

const ProfileModelDetails = ({ user }) => {
  const { characteristics, measurements } = user.modelInfo;
  const { eyes, skin, hair } = characteristics;
  const {
    chest, waist, hips, height, weight,
  } = measurements;
  return (
    <div className="profile-model-details">
      <h2>Characteristics</h2>
      <ul className="characteristics">
        <li>
            Eyes
          <span>{eyes}</span>
        </li>
        <li>
            Hair
          <span>{hair}</span>
        </li>
        <li>
            Skin
          <span>{skin}</span>
        </li>
      </ul>
      <h2>Measurements</h2>
      <ul className="measurements">
        <div>
          <li>
            Height
            <span>{height}</span>
          </li>
          <li>
            Weight
            <span>{weight}</span>
          </li>
        </div>
        <div>
          <li>
        Chest
            <span>{chest}</span>
          </li>
          <li>
            Waist
            <span>
              {waist}
            </span>
          </li>
          <li>
        Hips
            <span>{hips}</span>
          </li>
        </div>
      </ul>
      <p className="disclosure">Measurements are in cm</p>
    </div>
  );
};

ProfileModelDetails.propTypes = {
  user: PropTypes.shape({}).isRequired,
};

export default ProfileModelDetails;
