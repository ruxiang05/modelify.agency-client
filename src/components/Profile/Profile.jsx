import React from 'react';
import PropTypes from 'prop-types';
import { Route } from 'react-router-dom';
import EditProfile from './EditProfile';
import { UserContext } from '../../contexts/userContext';
import PageHeader from '../PageHeader';
import { ReactComponent as UnknownUserIcon } from '../../assets/icons/user-circle.svg';
import { ReactComponent as ForwardIcon } from '../../assets/icons/arrow-right.svg';
import ProfileModelDetails from './ProfileModelDetails';
import '../../styles/profile.scss';
import api from '../../api';
import { getToken } from '../../auth';

class Profile extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      edit: false,
    };

    this.logout = this.logout.bind(this);
    this.toggleEdit = this.toggleEdit.bind(this);
    this.getAuthURL = this.getAuthURL.bind(this);
  }

  componentDidMount() {
    const { location, history } = this.props;
    const urlParams = new URLSearchParams(location.search);
    if (urlParams.has('code')) {
      const code = urlParams.get('code');
      api.googleCalendar.authorize(getToken(), { code }).then((res) => {
        const { message, token } = res;
        if (message) {
          history.replace('/profile');
          api.users.editProfile(getToken(), { googleCalendarAPIToken: token });
        }
      });
    }
  }

  getAuthURL() {
    api.googleCalendar.getAuthURL(getToken()).then((res) => {
      const { redirect } = res;
      if (redirect) {
        window.location.assign(redirect);
      }
    });
  }

  toggleEdit() {
    const { edit } = this.state;
    this.setState({ edit: !edit });
  }

  logout() {
    const { history } = this.props;
    const { handleLogout } = this.context;
    handleLogout();
    history.push('/login');
  }

  render() {
    const { edit } = this.state;
    const { user, updateUser } = this.context;
    const { name, phoneNumber, role } = user;
    const { history } = this.props;

    return (
      <React.Fragment>
        {edit ? (
          <Route to="/edit"><EditProfile toggleEdit={this.toggleEdit} {...user} updateUser={updateUser} history={history} /></Route>
        ) : (
          <React.Fragment>
            <PageHeader title="Profile" />
            <div className="page">
              <div className="profile-details">
                <UnknownUserIcon />
                <div>
                  <h1 className="profile-name">{name}</h1>
                  <p>{phoneNumber}</p>
                </div>
              </div>
              {role === 'model' && <ProfileModelDetails user={user} />}
              <div className="profile-actions">
                <button type="button" onClick={this.getAuthURL}>
                  Connect to Google Calendar
                  <ForwardIcon />
                </button>
                <button type="button" onClick={this.toggleEdit}>
                  Edit my profile
                  <ForwardIcon />
                </button>
                <button type="button" onClick={this.logout}>
                  Logout
                  <ForwardIcon />
                </button>
              </div>
            </div>
          </React.Fragment>
        )}
      </React.Fragment>
    );
  }
}
Profile.propTypes = {
  history: PropTypes.shape({}).isRequired,
  location: PropTypes.shape({}).isRequired,
};
Profile.contextType = UserContext;
export default Profile;
