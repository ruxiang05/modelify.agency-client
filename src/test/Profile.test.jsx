/* Uses methods and/or components from react, prop-types, history and enzyme */
import React from 'react';
import { shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import PropTypes from 'prop-types';
import Profile from '../components/Profile/Profile';

jest.mock('../api.js');
let wrapper;

const contextTypes = {
  user: PropTypes.object,
  handleLogout: PropTypes.func,
};

const history = createBrowserHistory({
  basename: '/',
  keyLength: 0,
});

const location = {
  search: {
    code: 'test',
  },
};

const context = {
  handleLogout: jest.fn(),
  user: {
    email: 'email@email.com',
    name: 'name',
    role: 'agent',
    phoneNumber: '123',
  },
};

beforeAll(() => {
  Profile.contextTypes = contextTypes;
});

beforeEach(() => {
  wrapper = shallow(<Profile history={history} location={location} />, {
    context,
  });
});

describe('Profile', () => {
  it('should render Profile', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should toggle edit', () => {
    wrapper.instance().toggleEdit();
    expect(wrapper.state().edit).toBeTruthy();
  });

  it('should logout user', () => {
    wrapper.instance().logout();
    expect(context.handleLogout).toHaveBeenCalled();
    expect(history.location.pathname).toBe('/login');
  });

  it('should redirect to google auth', async () => {
    window.location.assign = jest.fn();
    await wrapper.instance().getAuthURL();
    expect(window.location.assign).toHaveBeenCalled();
  });
});
