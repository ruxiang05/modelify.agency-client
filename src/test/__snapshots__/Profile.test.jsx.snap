// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render Profile 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    history={
      Object {
        "action": "REPLACE",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/profile",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={
      Object {
        "search": Object {
          "code": "test",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <React.Fragment>
        <PageHeader
          backButton={false}
          title="Profile"
        />
        <div
          className="page"
        >
          <div
            className="profile-details"
          >
            <div
              className="profile-basic-details"
            >
              <ForwardRef />
              <div>
                <h1
                  className="profile-name"
                >
                  name
                </h1>
                <p>
                  123
                </p>
              </div>
            </div>
          </div>
          <div
            className="profile-actions"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              Connect to Google Calendar
              <ForwardRef />
            </button>
            <button
              onClick={[Function]}
              type="button"
            >
              Edit my profile
              <ForwardRef />
            </button>
            <button
              onClick={[Function]}
              type="button"
            >
              Logout
              <ForwardRef />
            </button>
          </div>
        </div>
      </React.Fragment>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <PageHeader
            backButton={false}
            title="Profile"
          />,
          <div
            className="page"
          >
            <div
              className="profile-details"
            >
              <div
                className="profile-basic-details"
              >
                <ForwardRef />
                <div>
                  <h1
                    className="profile-name"
                  >
                    name
                  </h1>
                  <p>
                    123
                  </p>
                </div>
              </div>
            </div>
            <div
              className="profile-actions"
            >
              <button
                onClick={[Function]}
                type="button"
              >
                Connect to Google Calendar
                <ForwardRef />
              </button>
              <button
                onClick={[Function]}
                type="button"
              >
                Edit my profile
                <ForwardRef />
              </button>
              <button
                onClick={[Function]}
                type="button"
              >
                Logout
                <ForwardRef />
              </button>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "action": undefined,
            "actionAltText": undefined,
            "actionIcon": undefined,
            "backButton": false,
            "goBackAction": undefined,
            "history": undefined,
            "title": "Profile",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="profile-details"
              >
                <div
                  className="profile-basic-details"
                >
                  <ForwardRef />
                  <div>
                    <h1
                      className="profile-name"
                    >
                      name
                    </h1>
                    <p>
                      123
                    </p>
                  </div>
                </div>
              </div>,
              <div
                className="profile-actions"
              >
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Connect to Google Calendar
                  <ForwardRef />
                </button>
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Edit my profile
                  <ForwardRef />
                </button>
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Logout
                  <ForwardRef />
                </button>
              </div>,
            ],
            "className": "page",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="profile-basic-details"
                  >
                    <ForwardRef />
                    <div>
                      <h1
                        className="profile-name"
                      >
                        name
                      </h1>
                      <p>
                        123
                      </p>
                    </div>
                  </div>,
                  false,
                ],
                "className": "profile-details",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <ForwardRef />,
                      <div>
                        <h1
                          className="profile-name"
                        >
                          name
                        </h1>
                        <p>
                          123
                        </p>
                      </div>,
                    ],
                    "className": "profile-basic-details",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <h1
                            className="profile-name"
                          >
                            name
                          </h1>,
                          <p>
                            123
                          </p>,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "name",
                            "className": "profile-name",
                          },
                          "ref": null,
                          "rendered": "name",
                          "type": "h1",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "123",
                          },
                          "ref": null,
                          "rendered": "123",
                          "type": "p",
                        },
                      ],
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                false,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    onClick={[Function]}
                    type="button"
                  >
                    Connect to Google Calendar
                    <ForwardRef />
                  </button>,
                  <button
                    onClick={[Function]}
                    type="button"
                  >
                    Edit my profile
                    <ForwardRef />
                  </button>,
                  <button
                    onClick={[Function]}
                    type="button"
                  >
                    Logout
                    <ForwardRef />
                  </button>,
                ],
                "className": "profile-actions",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Connect to Google Calendar",
                      <ForwardRef />,
                    ],
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Connect to Google Calendar",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                    },
                  ],
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Edit my profile",
                      <ForwardRef />,
                    ],
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Edit my profile",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                    },
                  ],
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Logout",
                      <ForwardRef />,
                    ],
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Logout",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                    },
                  ],
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <React.Fragment>
          <PageHeader
            backButton={false}
            title="Profile"
          />
          <div
            className="page"
          >
            <div
              className="profile-details"
            >
              <div
                className="profile-basic-details"
              >
                <ForwardRef />
                <div>
                  <h1
                    className="profile-name"
                  >
                    name
                  </h1>
                  <p>
                    123
                  </p>
                </div>
              </div>
            </div>
            <div
              className="profile-actions"
            >
              <button
                onClick={[Function]}
                type="button"
              >
                Connect to Google Calendar
                <ForwardRef />
              </button>
              <button
                onClick={[Function]}
                type="button"
              >
                Edit my profile
                <ForwardRef />
              </button>
              <button
                onClick={[Function]}
                type="button"
              >
                Logout
                <ForwardRef />
              </button>
            </div>
          </div>
        </React.Fragment>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <PageHeader
              backButton={false}
              title="Profile"
            />,
            <div
              className="page"
            >
              <div
                className="profile-details"
              >
                <div
                  className="profile-basic-details"
                >
                  <ForwardRef />
                  <div>
                    <h1
                      className="profile-name"
                    >
                      name
                    </h1>
                    <p>
                      123
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="profile-actions"
              >
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Connect to Google Calendar
                  <ForwardRef />
                </button>
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Edit my profile
                  <ForwardRef />
                </button>
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Logout
                  <ForwardRef />
                </button>
              </div>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "action": undefined,
              "actionAltText": undefined,
              "actionIcon": undefined,
              "backButton": false,
              "goBackAction": undefined,
              "history": undefined,
              "title": "Profile",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="profile-details"
                >
                  <div
                    className="profile-basic-details"
                  >
                    <ForwardRef />
                    <div>
                      <h1
                        className="profile-name"
                      >
                        name
                      </h1>
                      <p>
                        123
                      </p>
                    </div>
                  </div>
                </div>,
                <div
                  className="profile-actions"
                >
                  <button
                    onClick={[Function]}
                    type="button"
                  >
                    Connect to Google Calendar
                    <ForwardRef />
                  </button>
                  <button
                    onClick={[Function]}
                    type="button"
                  >
                    Edit my profile
                    <ForwardRef />
                  </button>
                  <button
                    onClick={[Function]}
                    type="button"
                  >
                    Logout
                    <ForwardRef />
                  </button>
                </div>,
              ],
              "className": "page",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="profile-basic-details"
                    >
                      <ForwardRef />
                      <div>
                        <h1
                          className="profile-name"
                        >
                          name
                        </h1>
                        <p>
                          123
                        </p>
                      </div>
                    </div>,
                    false,
                  ],
                  "className": "profile-details",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <ForwardRef />,
                        <div>
                          <h1
                            className="profile-name"
                          >
                            name
                          </h1>
                          <p>
                            123
                          </p>
                        </div>,
                      ],
                      "className": "profile-basic-details",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <h1
                              className="profile-name"
                            >
                              name
                            </h1>,
                            <p>
                              123
                            </p>,
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "name",
                              "className": "profile-name",
                            },
                            "ref": null,
                            "rendered": "name",
                            "type": "h1",
                          },
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "123",
                            },
                            "ref": null,
                            "rendered": "123",
                            "type": "p",
                          },
                        ],
                        "type": "div",
                      },
                    ],
                    "type": "div",
                  },
                  false,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      onClick={[Function]}
                      type="button"
                    >
                      Connect to Google Calendar
                      <ForwardRef />
                    </button>,
                    <button
                      onClick={[Function]}
                      type="button"
                    >
                      Edit my profile
                      <ForwardRef />
                    </button>,
                    <button
                      onClick={[Function]}
                      type="button"
                    >
                      Logout
                      <ForwardRef />
                    </button>,
                  ],
                  "className": "profile-actions",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Connect to Google Calendar",
                        <ForwardRef />,
                      ],
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Connect to Google Calendar",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                    ],
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Edit my profile",
                        <ForwardRef />,
                      ],
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Edit my profile",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                    ],
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Logout",
                        <ForwardRef />,
                      ],
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Logout",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                      },
                    ],
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "handleLogout": [MockFunction],
      "user": Object {
        "email": "email@email.com",
        "name": "name",
        "phoneNumber": "123",
        "role": "agent",
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
